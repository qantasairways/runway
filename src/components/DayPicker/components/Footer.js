import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'emotion';
import { Transition } from 'react-transition-group';
import {
  colours,
  layout,
  fontSize,
  fontFamily,
  fontWeight,
  mq
} from '../../../theme/airways';
import Button from '../../Button';

const wrapperStyle = {
  transform: 'translateY(100%)',
  transition: 'transform 300ms linear',
  width: '100%',
  position: 'fixed',
  bottom: 0,
  backgroundColor: colours.white,
  fontFamily: fontFamily.main
};

const wrapperTransitionStyle = {
  entering: {
    transform: 'translateY(0)'
  },
  entered: {
    transform: 'translateY(0)'
  },
  exiting: {
    transform: 'translateY(100%)'
  },
  exited: { transform: 'translateY(100%)' }
};

const innerWrapperStyle = {
  padding: `0 ${layout.gutter}`,
  width: '100%',
  maxWidth: '1030px',
  margin: '0 auto'
};

const preFooterStyle = {
  transition: 'transform 300ms linear',
  display: 'flex',
  transform: 'translateY(0)',
  flexDirection: 'column',
  justifyContent: 'center',
  padding: '11px 0',
  boxShadow: '0 -4px 6px 0 rgba(0, 0, 0, 0.08)',
  position: 'absolute',
  width: '100%',
  top: 0,
  left: 0,
  backgroundColor: colours.white,
  [mq.medium]: {
    flexDirection: 'row',
    justifyContent: 'left',
    alignItems: 'center'
  }
};

const preFooterTransitionStyle = {
  entering: {
    transform: 'translateY(-100%)'
  },
  entered: {
    transform: 'translateY(-100%)'
  },
  exiting: {
    transform: 'translateY(0)'
  },
  exited: { transform: 'translateY(0)' }
};

const preFooterTextStyle = {
  label: 'runway-footer_info-label',
  fontSize: '0.875rem',
  lineHeight: '2',
  color: colours.darkerGrey,
  [mq.medium]: {
    marginRight: '5px'
  }
};

const bottomFooterStyle = {
  padding: layout.gutter,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-between',
  position: 'relative'
};

const bottomFooterLeftCol = {
  display: 'flex',
  flexDirection: 'column',
  [mq.small]: {
    paddingRight: layout.gutter
  },
  [mq.medium]: {
    flexDirection: 'row',
    alignItems: 'center'
  }
};

const bottomFooterTextStyle = {
  label: 'runway-footer_summary-label',
  fontSize: fontSize.body,
  lineHeight: 1.56,
  [mq.medium]: {
    marginRight: '5px'
  }
};

const disclaimerStyle = {
  fontSize: '0.875rem',
  lineHeight: 1.29,
  color: colours.darkGrey
};

export function topDisclaimerStyle() {
  return css({
    label: 'runway-footer_disclaimer-label',
    ...disclaimerStyle
  });
}

// Not sure if this is needed any more?? Disclaimer will always stay on top?
export function bottonDisclaimerStyle() {
  return css({
    label: 'runway-footer_button_disclaimer-label',
    ...disclaimerStyle,
    [mq.medium]: {
      fontSize: fontSize.body
    }
  });
}

const actionButtonStyle = {
  fontWeight: fontWeight.regular,
  letterSpacing: '1.5px'
};

const Footer = ({
  summaryLabel,
  info,
  bottomFooterDisclaimer,
  preFooterDisclaimer,
  actionText,
  points,
  isCheapest,
  showPreFooter,
  showBottomFooter,
  onActionButtonClick
}) => (
  <Transition in={showBottomFooter} timeout={300}>
    {wrapperTransition => (
      <div
        css={{
          ...wrapperStyle,
          ...wrapperTransitionStyle[wrapperTransition]
        }}
      >
        <Transition in={info && showPreFooter} timeout={300}>
          {preFooterTransition => (
            <div
              css={{
                borderBottom:
                  showPreFooter && showBottomFooter
                    ? '1px solid #eeeeee'
                    : 'none',
                ...preFooterStyle,
                ...preFooterTransitionStyle[preFooterTransition]
              }}
            >
              <div css={innerWrapperStyle}>
                <span css={preFooterTextStyle}>{info}</span>
                {preFooterDisclaimer && (
                  <span css={topDisclaimerStyle()}>{preFooterDisclaimer}</span>
                )}
              </div>
            </div>
          )}
        </Transition>
        <div
          css={{
            ...innerWrapperStyle,
            ...bottomFooterStyle
          }}
        >
          <div css={bottomFooterLeftCol}>
            {summaryLabel && (
              <span css={bottomFooterTextStyle}>
                {summaryLabel}
                <span
                  css={{
                    color: isCheapest ? '#009400' : colours.darkerGrey,
                    fontWeight: fontWeight.bold
                  }}
                >
                  {points}
                </span>
              </span>
            )}
            {bottomFooterDisclaimer && (
              <span css={bottonDisclaimerStyle()}>
                {bottomFooterDisclaimer}
              </span>
            )}
          </div>
          <div>
            <Button
              label={actionText}
              cssOverrides={[actionButtonStyle]}
              onClick={onActionButtonClick}
            />
          </div>
        </div>
      </div>
    )}
  </Transition>
);

Footer.propTypes = {
  isCheapest: PropTypes.bool,
  showPreFooter: PropTypes.bool,
  showBottomFooter: PropTypes.bool,
  summaryLabel: PropTypes.string,
  points: PropTypes.string,
  info: PropTypes.string,
  preFooterDisclaimer: PropTypes.string,
  bottomFooterDisclaimer: PropTypes.string,
  actionText: PropTypes.string.isRequired,
  onActionButtonClick: PropTypes.func.isRequired
};

Footer.defaultProps = {
  isCheapest: false,
  showPreFooter: false,
  showBottomFooter: false,
  summaryLabel: '',
  points: '',
  info: '',
  preFooterDisclaimer: '',
  bottomFooterDisclaimer: ''
};

export default Footer;
