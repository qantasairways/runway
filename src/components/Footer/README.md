```js
<Footer
  preFooterInfo="Total return pricez in AUD per adult for a 25 day trip"
  bottomFooterDisclaimer=" + $344.90 ^"
  preFooterDisclaimer="^ taxes fees and carrier charges. Limited avaliability"
  actionText="Confirm"
  onActionButtonClick={() => {}}
  endDateData={{
    price: { value: 15127.0, isLowestPrice: false },
    currencySymbol: ' $'
  }}
/>
```
