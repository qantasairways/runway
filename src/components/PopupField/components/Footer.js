import React from 'react';
import PropTypes from 'prop-types';
import { generate } from 'shortid';

import { colours, fontFamily, fontWeight } from '../../../theme/airways';

import Button from '../../Button';

const textTruncationStyles = {
  whiteSpace: 'nowrap',
  overflow: 'hidden',
  textOverflow: 'ellipsis'
};

const primaryLabelContainerStyles = {
  display: 'flex',
  flexDirection: 'column',
  marginRight: '15px'
};

const primaryLabelStyles = {
  color: colours.darkerGrey,
  fontFamily: fontFamily.body,
  fontSize: '18px',
  ...textTruncationStyles
};

// eslint-disable-next-line react/prop-types
const buildLabel = ({ primaryLabels }) => {
  if (primaryLabels) {
    return (
      <div css={primaryLabelContainerStyles}>
        {primaryLabels.map((label, idx) => (
          <div
            // eslint-disable-next-line react/no-array-index-key
            key={idx}
            css={primaryLabelStyles}
          >
            {label}
          </div>
        ))}
      </div>
    );
  }
  return null;
};

const seperatorStyles = {
  borderTop: '1px solid #eeeeee'
};
const spacingStyles = { padding: '15px' };

const preFooterStyles = {
  ...spacingStyles,
  ...seperatorStyles,
  paddingTop: '9px',
  paddingBottom: '9px'
};

const buildPreFooter = preFooter =>
  preFooter && <div css={preFooterStyles}>{preFooter}</div>;

const actionButtonStyles = {
  fontFamily: 'Ciutadella',
  fontSize: '16px',
  fontWeight: fontWeight.regular,
  lineHeight: '1.5',
  letterSpacing: '1.5px',
  textAlign: 'center',
  height: '48px',
  width: '100px',
  padding: '0px 0px 0px 1px'
};

const primaryLabelOuterContainerStyles = {
  flex: 1,
  justifySelf: 'left',
  paddingRight: '15px',
  overflow: 'hidden'
};

const footerOuterContainerStyles = {
  label: 'runway-footer__root',
  zIndex: 1
};

const footerContainerStyles = {
  label: 'runway-footer__layout-container',
  ...spacingStyles,
  ...seperatorStyles,
  fontFamily: fontFamily.body,
  fontWeight: fontWeight.regular,
  background: colours.white,
  color: colours.darkerGrey,
  paddingRight: '15px',
  display: 'flex',
  justifyContent: 'space-between',
  alignItems: 'center'
};

const Footer = ({
  primaryLabels,
  primaryLabelAriaTitle,
  actionText,
  onAction,
  closeAriaLabel,
  preFooter
}) => {
  const summaryId = generate();

  return (
    <div css={footerOuterContainerStyles}>
      {buildPreFooter(preFooter)}
      <div css={footerContainerStyles}>
        <span
          title={primaryLabelAriaTitle}
          // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
          tabIndex={0}
          id={summaryId}
          css={primaryLabelOuterContainerStyles}
        >
          {buildLabel({ primaryLabels })}
        </span>
        <Button
          className="runway-footer__button"
          aria-label={closeAriaLabel}
          label={actionText}
          onClick={onAction}
          cssOverrides={[actionButtonStyles]}
        />
      </div>
    </div>
  );
};

Footer.propTypes = {
  primaryLabels: PropTypes.arrayOf(PropTypes.string).isRequired,
  primaryLabelAriaTitle: PropTypes.string.isRequired,
  actionText: PropTypes.string.isRequired,
  onAction: PropTypes.func,
  closeAriaLabel: PropTypes.string.isRequired,
  preFooter: PropTypes.node
};

Footer.defaultProps = {
  onAction: () => {},
  preFooter: null
};

export default Footer;
