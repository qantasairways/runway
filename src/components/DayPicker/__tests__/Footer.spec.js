import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../components/Footer';

describe('Footer', () => {
  let footer;

  it('renders correctly with defaults', () => {
    footer = shallow(<Footer actionText="" onActionButtonClick={e => e} />);

    expect(footer).toMatchSnapshot();
  });

  it('renders correctly with props provided', () => {
    footer = shallow(
      <Footer
        showPreFooter={false}
        showBottomFooter={false}
        bottomFootersummaryLabel="From"
        preFooterInfo="Lowest economy price per adult in AUD for a return trip."
        preFooterDisclaimer="^ Taxes, fees and carrier charges. Limited avaliability"
        bottomFooterDisclaimer="+ $34.90^"
        actionText="CONFIRM"
        onActionButtonClick={e => e}
        endDateData={null}
      />
    );

    expect(footer).toMatchSnapshot();
  });
});
