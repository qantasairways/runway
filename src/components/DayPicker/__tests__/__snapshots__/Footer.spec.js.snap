// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders only confirm button when endDateData does NOT include price.value 1`] = `
.emotion-2 {
  padding: 15px;
  width: 100%;
  max-width: 1030px;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media only screen and (min-width:321px) {
  .emotion-0 {
    padding-right: 15px;
  }
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

.emotion-1 {
  background-color: #ee0000;
  color: #fff;
  font-family: "Ciutadella",sans-serif;
  font-size: 1rem;
  width: 100%;
  height: 3.125rem;
  border-radius: 4px;
  border-width: 0;
  padding: 14px 24px 15px;
  font-weight: 700;
  line-height: 1.56;
  -webkit-letter-spacing: 0.1em;
  -moz-letter-spacing: 0.1em;
  -ms-letter-spacing: 0.1em;
  letter-spacing: 0.1em;
  text-transform: uppercase;
  cursor: pointer;
  font-weight: 400;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  height: 48px;
  width: 100px;
  border-radius: 4px;
  border-width: 0px;
  padding: 0 0 0 1px;
}

.emotion-1:hover {
  background-color: #ba0000;
}

.emotion-1:active {
  background-color: #ba0000;
}

.emotion-1:active {
  -webkit-transform: translate(1px,1px);
  -ms-transform: translate(1px,1px);
  transform: translate(1px,1px);
}

.emotion-3 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 300ms linear;
  -webkit-transition: transform 300ms linear;
  transition: transform 300ms linear;
  width: 100%;
  position: fixed;
  bottom: 0;
  background-color: #fff;
  font-family: "Ciutadella",sans-serif;
  box-shadow: 0 -4px 6px 0 rgba(0,0,0,0.08);
}

<Footer
  actionText="CONFIRM"
  bottomFooterDisclaimer="+ $34.90^"
  bottomFootersummaryLabel="From "
  endDateData={
    Object {
      "currencySymbol": "€",
      "price": null,
    }
  }
  onActionButtonClick={[Function]}
  pointsLabel=""
  preFooterDisclaimer="^ Taxes, fees and carrier charges. Limited avaliability"
  preFooterInfo="Lowest economy price per adult in AUD for a return trip."
  priceInPoints={false}
  showBottomFooter={true}
  showPreFooter={true}
>
  <Transition
    appear={false}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <div
      className="emotion-3"
    >
      <div
        className="emotion-2"
      >
        <div
          className="emotion-0"
        />
        <div>
          <Button
            className={null}
            cssOverrides={
              Array [
                Object {
                  "borderRadius": "4px",
                  "borderWidth": "0",
                  "fontWeight": 400,
                  "height": "48px",
                  "letterSpacing": "1.5px",
                  "padding": "0 0 0 1px",
                  "width": "100px",
                },
              ]
            }
            label="CONFIRM"
            onClick={[Function]}
            variant="solid"
          >
            <div
              className={null}
            >
              <button
                className="emotion-1"
                onClick={[Function]}
                type="button"
              >
                CONFIRM
              </button>
            </div>
          </Button>
        </div>
      </div>
    </div>
  </Transition>
</Footer>
`;

exports[`Footer renders preFooterDisclaimer and bottom footer text contents when endDateData includes price.value 1`] = `
.emotion-8 {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
  -webkit-transition: -webkit-transform 300ms linear;
  -webkit-transition: transform 300ms linear;
  transition: transform 300ms linear;
  width: 100%;
  position: fixed;
  bottom: 0;
  background-color: #fff;
  font-family: "Ciutadella",sans-serif;
}

.emotion-7 {
  padding: 15px;
  width: 100%;
  max-width: 1030px;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media only screen and (min-width:321px) {
  .emotion-5 {
    padding-right: 15px;
  }
}

@media only screen and (min-width:768px) {
  .emotion-5 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

.emotion-4 {
  font-size: 1.125rem;
  line-height: 1.56;
}

.emotion-3 {
  color: #323232;
  font-weight: 700;
  padding: 0 5px;
}

.emotion-2 {
  border-bottom: 1px solid #eeeeee;
  -webkit-transition: -webkit-transform 300ms linear;
  -webkit-transition: transform 300ms linear;
  transition: transform 300ms linear;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transform: translateY(-100%);
  -ms-transform: translateY(-100%);
  transform: translateY(-100%);
  box-shadow: 0 -4px 6px 0 rgba(0,0,0,0.08);
  width: 100%;
  background-color: #fff;
}

.emotion-1 {
  padding: 7.5px 15px;
  width: 100%;
  max-width: 1030px;
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.emotion-0 {
  font-size: 0.875rem;
  line-height: 2;
  color: #323232;
}

@media only screen and (min-width:768px) {
  .emotion-0 {
    margin-right: 5px;
  }
}

.emotion-6 {
  background-color: #ee0000;
  color: #fff;
  font-family: "Ciutadella",sans-serif;
  font-size: 1rem;
  width: 100%;
  height: 3.125rem;
  border-radius: 4px;
  border-width: 0;
  padding: 14px 24px 15px;
  font-weight: 700;
  line-height: 1.56;
  -webkit-letter-spacing: 0.1em;
  -moz-letter-spacing: 0.1em;
  -ms-letter-spacing: 0.1em;
  letter-spacing: 0.1em;
  text-transform: uppercase;
  cursor: pointer;
  font-weight: 400;
  -webkit-letter-spacing: 1.5px;
  -moz-letter-spacing: 1.5px;
  -ms-letter-spacing: 1.5px;
  letter-spacing: 1.5px;
  height: 48px;
  width: 100px;
  border-radius: 4px;
  border-width: 0px;
  padding: 0 0 0 1px;
}

.emotion-6:hover {
  background-color: #ba0000;
}

.emotion-6:active {
  background-color: #ba0000;
}

.emotion-6:active {
  -webkit-transform: translate(1px,1px);
  -ms-transform: translate(1px,1px);
  transform: translate(1px,1px);
}

<Footer
  actionText="CONFIRM"
  bottomFooterDisclaimer="+ $34.90^"
  bottomFootersummaryLabel="From "
  endDateData={
    Object {
      "currencySymbol": "€",
      "price": Object {
        "value": 3000,
      },
    }
  }
  onActionButtonClick={[Function]}
  pointsLabel=""
  preFooterDisclaimer="^ Taxes, fees and carrier charges. Limited avaliability"
  preFooterInfo="Lowest economy price per adult in AUD for a return trip."
  priceInPoints={false}
  showBottomFooter={true}
  showPreFooter={true}
>
  <Transition
    appear={false}
    enter={true}
    exit={true}
    in={true}
    mountOnEnter={false}
    onEnter={[Function]}
    onEntered={[Function]}
    onEntering={[Function]}
    onExit={[Function]}
    onExited={[Function]}
    onExiting={[Function]}
    timeout={300}
    unmountOnExit={false}
  >
    <div
      className="emotion-8"
    >
      <Transition
        appear={false}
        enter={true}
        exit={true}
        in={true}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={300}
        unmountOnExit={false}
      >
        <div
          className="emotion-2"
        >
          <div
            className="emotion-1"
          >
            <span
              className="emotion-0"
            >
              Lowest economy price per adult in AUD for a return trip.
            </span>
          </div>
        </div>
      </Transition>
      <div
        className="emotion-7"
      >
        <div
          className="emotion-5"
        >
          <span
            className="emotion-4"
          >
            From 
            <span
              className="emotion-3"
            >
              <span>
                €3,000.00
              </span>
            </span>
          </span>
        </div>
        <div>
          <Button
            className={null}
            cssOverrides={
              Array [
                Object {
                  "borderRadius": "4px",
                  "borderWidth": "0",
                  "fontWeight": 400,
                  "height": "48px",
                  "letterSpacing": "1.5px",
                  "padding": "0 0 0 1px",
                  "width": "100px",
                },
              ]
            }
            label="CONFIRM"
            onClick={[Function]}
            variant="solid"
          >
            <div
              className={null}
            >
              <button
                className="emotion-6"
                onClick={[Function]}
                type="button"
              >
                CONFIRM
              </button>
            </div>
          </Button>
        </div>
      </div>
    </div>
  </Transition>
</Footer>
`;

exports[`Footer renders with defaults 1`] = `
<Transition
  appear={false}
  enter={true}
  exit={true}
  in={false}
  mountOnEnter={false}
  onEnter={[Function]}
  onEntered={[Function]}
  onEntering={[Function]}
  onExit={[Function]}
  onExited={[Function]}
  onExiting={[Function]}
  timeout={300}
  unmountOnExit={false}
>
  <Component />
</Transition>
`;

exports[`Footer renders with props provided 1`] = `
<Transition
  appear={false}
  enter={true}
  exit={true}
  in={false}
  mountOnEnter={false}
  onEnter={[Function]}
  onEntered={[Function]}
  onEntering={[Function]}
  onExit={[Function]}
  onExited={[Function]}
  onExiting={[Function]}
  timeout={300}
  unmountOnExit={false}
>
  <Component />
</Transition>
`;
